src/components/validations/validation-icon.component.js

```diff
@@ -32,6 +33,7 @@ const ValidationIcon = ({
     hasFocus: groupHasFocus,
     hasMouseOver: groupHasMouseOver,
   } = useContext(InputGroupContext);
+  const [hasHover, setHasHover] = React.useState(false);
```

Comment:
Why do we need `hasHover` when we have `hasMouseOver` from the `InputContext`?

Reply:
`hasMouseOver` is from the context for the input, the tooltip needs to also trigger when the `ValidationIcon` has a `mouseOver` or `focus` event. I think the naming it `hasHover, setHasHover` is really poor on my part as it doesn't really reflect what it's doing, I'll rename it to something like `internalTooltipTrigger` and `setInternalTooltipTrigger`